src/main/java/edu/cornell/kfs/module/purap/document/service/impl/CuB2BShoppingServiceImpl.java

```diff
@@ -312,8 +356,9 @@ public String getPunchOutSetupRequestMessage(Person user, B2BInformation b2bInfo
         // KFSPTS-1720
         cxml.append("      <Extrinsic name=\"FirstName\">").append(user.getFirstName()).append("</Extrinsic>\n");
         cxml.append("      <Extrinsic name=\"LastName\">").append(user.getLastName()).append("</Extrinsic>\n");
-        cxml.append("      <Extrinsic name=\"Role\">").append(getPreAuthValue(user.getPrincipalId())).append("</Extrinsic>\n");
```

Comment:
I suggest doing roles the "old" way for the roleSet is eShop.  Otherwise use jaggaerRoleService get a list of roles, and then loop as you have coded below

Reply:
I'll rework this to implement the old/new split like you had suggested.